tests/updateDBTest.php

```diff
@@ -46,6 +49,42 @@ public function testUpdateDBStatus() {
     $this->assertEquals('This is a test of the emergency broadcast system.', trim($out->description));
   }
 
+  /**
+   * Tests that updatedb command returns properly a failure.
+   */
+  function  testFailedUpdate() {
+    $sites = $this->setUpDrupal(1, TRUE);
+    $options = [
+      'yes' => NULL,
+      'root' => $root = $this->webroot(),
+      'uri' => key($sites),
+    ];
+    $this->setupModulesForTests($root);
+
+    $this->drush('pm-enable', ['woot'], $options);
+
+    // Force re-run of the post-update woot_post_update_failing().
+    $this->drush('sql:query', ["SELECT value FROM key_value WHERE collection = 'post_update' AND name = 'existing_updates'"], $options);
+    $functions = unserialize($this->getOutput());
+    unset($functions[array_search('woot_post_update_failing', $functions)]);
+    $functions = serialize($functions);
+    $this->drush('sql:query', ["UPDATE key_value SET value = '$functions' WHERE collection = 'post_update' AND name = 'existing_updates'"], $options);
+
+    // Run updates. woot_post_update_failing() is failing.
+    $return = $this->drush('updatedb', [], $options);
```

Comment:
Thanks!

$this->drush() asserts for a zero exit code. Once we are actually returning wrong code, we will have to pass a param at the end - \Unish\CommandUnishTestCase::EXIT_ERROR